layer/layer.go

```diff
@@ -188,6 +188,8 @@ type Store interface {
 	GetRWLayer(id string) (RWLayer, error)
 	GetMountID(id string) (string, error)
 	ReleaseRWLayer(RWLayer) ([]Metadata, error)
+	LockRWLayer()
+	UnlockRWLayer()
```

Comment:
@cpuguy83  `daemon.layerStore.ReleaseRWLayer(container.RWLayer)` has already been locked with the same lock.